lib/modules/download/requesters/search.rb

```diff
@@ -16,19 +16,19 @@ def domain
     'search'
   end
 
-  private
+
 
   def identifier
     token
   end
 
   def token
     @token ||= begin
-      filters_dump = Marshal.dump filters.keys.sort
+      filters_dump = Marshal.dump filters.sort.to_json
```

Comment:
This works well and it seems to fix the issue!
However, I'm wondering if:

1. We could fix the `sort` deprecation warning and to something like `filters.to_hash.sort.to_json` to fix it.
2. Just use values instead, like `filters.values.sort`. I don't think there will be values which are the same across the different filters, but we could just go for option 1 to be extra safe.